sdv/single_table/base.py

```diff
@@ -108,6 +109,15 @@ def _check_metadata_updated(self):
                 ' in future SDV versions.'
             )
 
+    def _validate_regex_format(self):
+        if self.metadata.tables:
+            id_columns = self.metadata.get_column_names(table_name=self._table_name, sdtype='id')
+            for column_name in id_columns:
+                regex = (
+                    self.metadata.tables[self._table_name].columns[column_name].get('regex_format')
+                )
+                _check_regex_format(self._table_name, column_name, regex)
```

Comment:
Does this work if you have an ID column without a regex format?

Reply:
Oh I see, we check for a regex in the `_check_regex_format` function. Can we add an ID column without a regex pattern to the unit/integration tests?

Reply:
Yes, done in [0c73bea](https://github.com/sdv-dev/SDV/pull/2441/commits/0c73bea310464d0b8e4b1626511770c3e5daa151)